Upgrade libusb from 0.x to 1.0

davidgfnet opened the pull request:
This should be fine on all platforms. libusb 1.0 has been the main libusb for a while now.
The old branch is only provided for old apps, usually through a compat layer.

Tested on a Linux box, windows builds look fine (need testing).

sharkwouter commented:
Before merging this, let me do some extensive tests on Windows. The last attempt from #16 was also fully working on Linux, but I wasn't able to get it to work on Windows at all. That is why we're still on libusb-0.1.

davidgfnet commented:
I think I found the issue :D The reason why it was working on non-win is the ifdef cygwin that sets a timeout. On windows we use a timeout and we were not using the right error code check.
I also noticed that the devel version of readline was switched yesterday (to use v8 by default), so my test build complains about the missing dll (this is only for pspsh though), so I updated and changed that too.

sharkwouter commented:
It might be better to use ldd to copy the libraries for Windows. I have an example here: https://github.com/sharkwouter/zt1-engine/blob/main/.github/workflows/build.yml#L67

sharkwouter reviewed (approved):
I just tested this on Windows and it works well. Thanks for fixing this long outstanding issues!
